bats/tests/profile/create-profile-output.bats

```diff
@@ -0,0 +1,454 @@
+<plist version="1.0">
+  <dict>
+    <key>containerEngine</key>
+    <dict>
+      <key>allowedImages</key>
+      <dict>
+        <key>patterns</key>
+        <array>
+          <string>abc</string>
+          <string>ghi</string>
+          <string>def</string>
+        </array>
+      </dict>
+    </dict>
+    <key>kubernetes</key>
+    <dict>
+      <key>enabled</key>
+      <false/>
+    </dict>
+    <key>WSL</key>
+    <dict>
+      <key>integrations</key>
+      <dict>
+        <key>first</key>
+        <true/>
+        <key>second</key>
+        <false/>
+      </dict>
+    </dict>
+  </dict>
+</plist>
+
+EOF
+}
+
+@test 'plist-encodes multi-string values and maps from a file' {
+    run rdctl create-profile --output plist --body "$(json_maps_and_lists)"
+    assert_complex_plist_output
+}
+
+@test 'plist-encodes multi-string values and maps from a json string' {
+    run rdctl create-profile --output plist --body "$(json_maps_and_lists)"
+    assert_complex_plist_output
+}
+
+# Actual output-testing of this input is done in `plist_test.go` -- the purpose of this test is to just
+# make sure that we're generating compliant data.
+
+@test 'verify converted special-char input is escaped and satisfies plutil' {
+    if ! is_macos; then
+        skip "Test requires the plist utility and only works on macOS"
+    fi
+    # This input form is ok here because it won't run in WSL/Windows
+    run rdctl create-profile --output plist --input <(json_with_special_chars)
+    assert_success
+    plutil -s - <<<"$output"
+}
+
+@test 'verify converted special-char output' {
+    run rdctl create-profile --output plist --body "$(json_with_special_chars)"
+    assert_success
+    assert_output - <<'END'
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+  <dict>
+    <key>application</key>
+    <dict>
+      <key>extensions</key>
+      <dict>
+        <key>allowed</key>
+        <dict>
+          <key>enabled</key>
+          <false/>
+          <key>list</key>
+          <array>
+            <string>less-than:&lt;</string>
+            <string>greater:&gt;</string>
+            <string>and:&amp;</string>
+            <string>d-quote:&#34;</string>
+            <string>emoji:😀</string>
+          </array>
+        </dict>
+        <key>installed</key>
+        <dict>
+          <key>key-with-ampersand: &amp;</key>
+          <true/>
+          <key>key-with-emoji: 🐤</key>
+          <false/>
+          <key>key-with-greater-than: &gt;</key>
+          <true/>
+          <key>key-with-less-than: &lt;</key>
+          <true/>
+        </dict>
+      </dict>
+    </dict>
+    <key>containerEngine</key>
+    <dict>
+      <key>name</key>
+      <string>small-less-&lt;-than</string>
+    </dict>
+  </dict>
+</plist>
+END
+}
+
+@test "and shutdown" {
+    if is_macos; then
+        rdctl shutdown
+    fi
+}
```

Comment:
This should not be necessary; we really should look into why this happens for you. Is this specific to using `RD_LOCATION=dev`? Because I always use `RD_LOCATION=system`, and it doesn't need a shutdown during teardown.

Will approve as is though and create a seoarate issue.